lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -3167,7 +3169,10 @@ TFGraphFunctionLowering::lowerConditionalRegion(ConditionalSESERegion *r) {
   //   .Attr("Tin: list(type) >= 0")
   //   .Attr("Tout: list(type) >= 0")
   auto opLocString = getUniqueName(loc, "op");
-  auto *op = TF_NewOperation(graphFn.getGraph(), "If", opLocString.c_str());
+  bool useSideEffectingIf = trueFnHasSideEffects || falseFnHasSideEffects;
```

Comment:
Both `trueFnHasSideEffects` and `falseFnHasSideEffects` are hard-coded false above. What would happen if there's a side-effecting operation in the body?

Reply:
They are output params of buildGraphFunction(). Does that clarify?

Reply:
Also, there are test cases that generate (the stateful) "If" and "While" ops